File: solutions/ruby/locomotive-engineer/1/locomotive_engineer.rb
# frozen_string_literal: true

class LocomotiveEngineer
  def self.generate_list_of_wagons(*wagons)
    wagons
  end

  def self.fix_list_of_wagons((first, second, one, *each_wagons_id), missing_wagons)
    [one, missing_wagons, each_wagons_id, first, second].flatten
  end

  def self.add_missing_stops(from_to = {}, stops = {})
    x, y = from_to.values_at(:from, :to)
    { from: x, to: y, stops: stops&.values || [] }
  end

  def self.extend_route_information(route, more_route_information)
    route.merge!(more_route_information)
  end
end

File: solutions/python/locomotive-engineer/1/locomotive_engineer.py
"""Functions which helps the locomotive engineer to keep track of the train."""


def get_list_of_wagons(*wagons):
    return list(wagons)


def fix_list_of_wagons(each_wagons_id, missing_wagons):
    first, second, one, *rest = each_wagons_id
    return [one, *missing_wagons, *rest, first, second]


def add_missing_stops(routing, **kwargs):
    stops = []
    for key, value in kwargs.items():
        stops.append(value)
    return {**routing, 'stops': stops}


def extend_route_information(route, more_route_information):
    return {**route, **more_route_information}


def fix_wagon_depot(wagons_rows):
    a, b, c = zip(*wagons_rows)
    return [[a[0], a[1], a[2]], [b[0], b[1], b[2]],[c[0], c[1], c[2]]]